python/ray/serve/_private/router.py

```diff
@@ -434,13 +442,19 @@ def update_replicas(self, replicas: List[ReplicaWrapper]):
         """
         new_replicas = {}
         new_replica_id_set = set()
-        new_colocated_replica_ids = set()
+        new_node_colocated_replica_ids = set()
+        new_az_colocated_replica_ids = set()
         new_multiplexed_model_id_to_replica_ids = defaultdict(set)
         for r in replicas:
             new_replicas[r.replica_id] = r
             new_replica_id_set.add(r.replica_id)
             if self._self_node_id is not None and r.node_id == self._self_node_id:
```

Comment:
When node_id be not available?

Reply:
In an actual Ray cluster I think node id should always be available since it's passed in through a handle. `PowerOfTwoChoicesReplicaScheduler` just takes it in as an optional parameter for now, not sure if there's a specific reason for that.

Reply:
Users can get a handle in the driver script, which doesn't run on a Ray node. The handle will contain a `Router`, but there won't be a `node_id`.

Reply:
that's incorrect on two counts:

1. driver script still has a node id (usually head node)
2. these IDs are for the replicas, not the driver

I believe `node_id` should always be populated